### PR TITLE
chore: add jsdoc to 26_fetch.js and enable some fetch tests

### PR DESCRIPTION
**Pull request by lucacasonato**
This adds some JS doc to 26_fetch.js, and fixes some issues.

### CODE REVIEW

File: op_crates/fetch/26_fetch.js [hunk 1]
```diff
@@ -1211,16 +1538,23 @@
...
     if (requestBodyRid) {
+      if (!(body instanceof ReadableStream)) {
+        throw new TypeError("Unreachable state (body is not ReadableStream).");
+      }
       const writer = new WritableStream({
+        /**
+       * 
+       * @param {Uint8Array} chunk 
+       * @param {WritableStreamDefaultController} controller 
+       */
```
Comment (ry) [inline op_crates/fetch/26_fetch.js:1552]: alignment off here

File: op_crates/fetch/26_fetch.js [hunk 2]
```diff
@@ -125,6 +175,10 @@
       return m;
     };
 
+    /** 
+     * @param {number} n
+    * @returns {void}
+    */
```
Comment (ry) [inline op_crates/fetch/26_fetch.js:181]: alignment

File: op_crates/fetch/26_fetch.js [hunk 3]
```diff
@@ -76,23 +110,39 @@
...
+     */
     write(p) {
       const n = this.writeSync(p);
       return Promise.resolve(n);
     }
 
+    /** 
+     * @param {number} n
+    * @returns {number}
+      */
```
Comment (ry) [inline op_crates/fetch/26_fetch.js:145]: alignment

File: op_crates/fetch/26_fetch.js [hunk 4]
```diff
@@ -1,5 +1,14 @@
 // Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
 
+// @ts-check
```
Comment (ry) [inline op_crates/fetch/26_fetch.js:3]: What does this do?

**Review by ry — APPROVED** [review]
LGTM modulo some superficial comments

File: op_crates/fetch/26_fetch.js (hunk 4 above)
Comment (lucacasonato) [reply op_crates/fetch/26_fetch.js:3]: Makes my VS Code scream at me if I use the wrong type in the wrong spot. We don't actually run TSC on the code in CI. Purely for development experience

**Pull request merged**